packages/compass-import-export/src/utils/collection-stream.ts

```diff
@@ -67,27 +136,29 @@ class WritableCollectionStream extends Writable {
 
     debug('draining buffered docs', this.batch.length);
 
-    this._executeBatch(callback);
+    void this._executeBatch(callback);
   }
 
-  async _executeBatch(callback) {
+  async _executeBatch(callback: (err?: Error) => void) {
     const documents = this.batch;
 
     this.batch = [];
 
-    let result;
-    let error;
+    let result: BulkOpResult;
 
     try {
-      result = await this._collection().bulkWrite(
-        documents.map((doc) => ({ insertOne: doc })),
+      result = await this.dataService.bulkWrite(
```

Comment:
definitely not for this PR, but how does this gets canceled? Would it be worth creating a follow-up ticket in the import/export epic?